aten/src/ATen/core/Scalar.h

```diff
@@ -50,15 +60,22 @@ class AT_API Scalar {
   bool isIntegral() const {
     return Tag::HAS_i == tag;
   }
+  bool isComplex() const {
+    return Tag::HAS_z == tag;
+  }
 
   Scalar operator-() const;
 
 private:
-  enum class Tag { HAS_d, HAS_i };
+  enum class Tag { HAS_d, HAS_i, HAS_z };
   Tag tag;
   union {
     double d;
-    int64_t i = 0;
+    int64_t i;
```

Comment:
glad that we don't have complex integers

Reply:
Gaussian integers oh yissss